server/svix-server/src/worker.rs

```diff
@@ -326,6 +330,33 @@ pub async fn worker_loop(
     }
 }
 
+/// Nullifies the payload column for expired messages
+async fn clean_expired_messages(
+    pool: &DatabaseConnection,
+) -> std::result::Result<UpdateResult, DbErr> {
+    message::Entity::update_many()
```

Comment:
This scans the whole DB on every run. Adding an index on this model may be too expensive to be worth it.
I wonder whether there's a different strategy, like adding a new model and then just deleting the whole "content" item when expired? Though that may be expensive too.

Any thoughts?

Reply:
Is your proposal to add a new table for storing payloads and their expirations, where we delete the entire row on expiration? I think that's reasonable, and brings the db scan from the number of every message to only the number of messages that still have payloads.